transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -128,6 +169,23 @@ impl SignatureRctBulletproofs {
             ));
         }
 
+        // pseudo output token ids must be provided if mixed transactions is enabled
+        if block_version.mixed_transactions_are_supported() {
+            if self.pseudo_output_commitments.len() != self.pseudo_output_token_ids.len() {
+                return Err(Error::MissingPseudoOutputTokenIds);
+            }
+            if output_commitments.len() != self.output_token_ids.len() {
+                return Err(Error::MissingOutputTokenIds);
+            }
+        } else {
```

Comment:
also check if `range_proofs` is empty?

Reply:
yeah that happens later right now, when validating the range proofs, but we could do all those checks here instead